base_classes/NXbeam.nxdl.xml

```diff
@@ -245,6 +317,87 @@
             <dim index="1" value="nP"/>
         </dimensions>
     </field>
+    <field name="pulse_energy" type="NX_FLOAT" units="NX_ENERGY">
+        <doc>
+             Energy of a single pulse at the diagnostic point
+        </doc>
+    </field>
+    <field name="average_power" type="NX_FLOAT" units="NX_POWER">
+        <doc>
+             Average power at the diagnostic point
+        </doc>
+    </field>
+    <field name="fluence" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             Incident fluence at the diagnostic point
+        </doc>
+        <attribute name="units" type="NX_CHAR">
+            <doc>
+                 Here: SI units are 'J/m2', customary 'mJ/cm2'.
+            </doc>
+        </attribute>
+    </field>
+    <field name="pulse_duration" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             FWHM duration of the pulses at the diagnostic point
+        </doc>
+    </field>
+    <field name="pulse_delay" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             Delay time between two pulses of a pulsed beam.
+        </doc>
+        <attribute name="reference_beam" type="NX_CHAR">
+            <doc>
+                 A reference to the beam in relation to which the delay is.
+            </doc>
+        </attribute>
+    </field>
+    <field name="frog_trace" type="NX_FLOAT">
+        <doc>
+             FROG trace of the pulse.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="nx"/>
+            <dim index="2" value="ny"/>
+        </dimensions>
+    </field>
+    <field name="frog_delays" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             Horizontal axis of a FROG trace, i.e. delay.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="nx"/>
+        </dimensions>
+    </field>
+    <field name="frog_frequencies" type="NX_FLOAT" units="NX_FREQUENCY">
+        <doc>
+             Vertical axis of a FROG trace, i.e. frequency.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ny"/>
+        </dimensions>
+    </field>
+    <field name="chirp_type" type="NX_CHAR">
+        <doc>
+             The type of chirp implemented
+        </doc>
+    </field>
+    <field name="chirp_GDD" type="NX_FLOAT" units="NX_TIME">
+        <doc>
+             Group delay dispersion of the pulse for linear chirp
+        </doc>
+    </field>
+    <field name="previous_device">
```

Comment:
The general way is place an `NXbeam` in a device in `NXinstrument` - whether this is good enough or a new way to chain devices together should be a NIAC decision.

Reply:
Your comment would be resolved explicitly by a NIAC vote.  IE whether we should allow previous/next device